real_intent/deliver/webhook.py

```
@@ -0,0 +1,73 @@
+    """Deliver leads to a webhook."""
+
+    def __init__(self, webhook_url: str | HttpUrl, per_lead_insights: dict[str, str] | None = None):
+        """Initialize the deliverer."""
+        if not isinstance(webhook_url, (str, HttpUrl)):
+            raise ValueError("Webhook URL must be a string or HttpUrl.")
+
+        if isinstance(webhook_url, str):
+            try:
+                webhook_url = HttpUrl(webhook_url)
+            except ValueError:
+                raise ValueError("Invalid webhook URL provided.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)